tdigest.go

```diff
@@ -219,12 +227,19 @@ func (t *TDigest) CDF(x float64) float64 {
 	return weightedAverage(t.cumulative[upper-1], z2, t.cumulative[upper], z1) / t.processedWeight
 }
 
-func (t *TDigest) integratedQ(k float64) float64 {
-	return (math.Sin(math.Min(k, t.Compression)*math.Pi/t.Compression-math.Pi/2.0) + 1.0) / 2.0
+type scaler interface {
```

Comment:
The implementation of `scaler` is exported on the `TDigetst` struct.  I think if you're going to export that, then the interface should be exported as well as it's methods.  Otherwise you should unexport `Scaler` on the `TDigest`.

Reply:
So the implementation (there will be more) are public K1, etc, and the field is public, i don't see any reason to make the external interface public (or its internal methods) when it's just an implementation detail. Thoughts?